feature-wallet-api/src/main/java/io/novafoundation/nova/feature_wallet_api/domain/validation/SufficientBalanceConsideringConsumersValidation.kt

```diff
@@ -0,0 +1,62 @@
+package io.novafoundation.nova.feature_wallet_api.domain.validation
+
+import io.novafoundation.nova.common.validation.Validation
+import io.novafoundation.nova.common.validation.ValidationStatus
+import io.novafoundation.nova.common.validation.ValidationSystemBuilder
+import io.novafoundation.nova.common.validation.valid
+import io.novafoundation.nova.common.validation.validOrError
+import io.novafoundation.nova.feature_wallet_api.data.network.blockhain.assets.AssetSourceRegistry
+import io.novafoundation.nova.feature_wallet_api.data.network.blockhain.assets.existentialDepositInPlanks
+import io.novafoundation.nova.feature_wallet_api.data.network.blockhain.assets.totalCanBeDroppedBelowMinimumBalance
+import io.novafoundation.nova.runtime.ext.isCommissionAsset
+import io.novafoundation.nova.runtime.multiNetwork.chain.model.Chain
+import java.math.BigInteger
+
+class SufficientBalanceConsideringConsumersValidation<P, E>(
+    private val assetSourceRegistry: AssetSourceRegistry,
+    private val chainExtractor: (P) -> Chain,
+    private val assetExtractor: (P) -> Chain.Asset,
+    private val totalBalanceExtractor: (P) -> BigInteger,
+    private val feeExtractor: (P) -> BigInteger,
+    private val amountExtractor: (P) -> BigInteger,
+    private val error: (P, existentialDeposit: BigInteger) -> E
+) : Validation<P, E> {
+
+    override suspend fun validate(value: P): ValidationStatus<E> {
+        val chain = chainExtractor(value)
+        val asset = assetExtractor(value)
+
+        val totalCanDropBelowMinimumBalance = assetSourceRegistry.totalCanBeDroppedBelowMinimumBalance(asset)
+
+        if (asset.isCommissionAsset && totalCanDropBelowMinimumBalance) {
```

Comment:
Seems like you're using `totalCanBeDroppedBelowMinimumBalance` as `total can NOT be dropped below minimum. It was like that in implementation and in all checks. Can you please revisit all places it is used and align usage with function name?

Reply:
Of course. I'll check it and test again